### DISCUSSION TITLE
Disable auto resize or draw inside rect/box

### DISCUSSION THREAD

**Discussion by Loewe1000** [cetz-package/cetz#432] [Q&A] [answered]
I would like to create some graphs for the topic of vector calculus.

Among other things, I would like to represent a straight line through two points in R^2.

I have already specified and drawn the two points:
<img width="400" alt="image" src="https://github.com/johannes-wolf/cetz/assets/1015274/f04c37fe-ffb6-46d3-b244-81bd5c452039">

Now I want to draw an (infinitely long) straight line through these two points that does not exceed the bounding box.

Do you have any idea how I can implement something like this? I would like to deactivate the automatic enlargement of the canvas and specify the size (and position) of the canvas.

Right now I am drawing the straight line with an "overshoot":
line((a, 2, b), (b, 2, a))

However, I don't want to calculate this with the boundaries of the bounding box.
I have already tried this, but gave up after more than 50 lines and hope that there is a simpler solution.

I am trying to recreate something like this:
<img width="784" alt="image" src="https://github.com/johannes-wolf/cetz/assets/1015274/674a110c-4b16-48d3-9674-46b54f1b6efb">

**Comment by johannes-wolf** [comment, marked as answer]
Are you using 0.1.2 or 0.2.0? Why is the overshoot not a good solution? You can also use absolute distances for that, using `(a: a, b: b, number: -1, abs: true)`

Example:
```typst
#import "@preview/cetz:0.1.2"
#cetz.canvas({
  import cetz.draw: *

  let a = (0,0)
  let b = (4,2)
  let c = (a: b, b: a, angle: 90deg, number: .5)
  
  set-style(circle: (radius: .1, fill: auto, stroke: auto),
            content: (padding: .3))
  
  line((a: a, b: b, number: -1, abs: true), 
       (a: b, b: a, number: -1, abs: true))
  line(b, c, stroke: red)

  cetz.angle.angle(b, a, c, label: $ dot $)
  
  fill(black)
  circle(a)
  circle(b)
  circle(c)

  content(a, $ g $, anchor: "bottom")
  content(b, $ F $, anchor: "bottom")
  content(c, $ P $, anchor: "left")
})
```

![grafik](https://github.com/johannes-wolf/cetz/assets/519002/44753303-f173-4740-a585-3f25513788cd)


If you are on 0.2.0, the implementation of `hide(..)` shows how to draw (or not) without a bounding box. You could copy the `hide` implementation but enable drawing.

<details>
<summary>Full Example</summary>

```typst
#block(stroke: green, fill: green.lighten(90%), {
  place(cetz.canvas({
    import cetz.draw: *
    content((0,0), text(white)[*Skalarprodukt*], name: "text", padding: .2)
    on-layer(-1, line("text.top-left", (rel: (.3, 0), to: "text.top-right"),
      "text.bottom-right", "text.bottom-left", close: true, fill: green, stroke: none))
  }))
  table(stroke: none, columns: 2,
    cetz.canvas({
      import cetz.draw: *
    
      let a = (0,0)
      let b = (4,2)
      let c = (a: b, b: a, angle: 90deg, number: .5)
      
      set-style(circle: (radius: .05, fill: auto, stroke: auto),
                content: (padding: .3))
      
      line((a: a, b: b, number: -1, abs: true), 
           (a: b, b: a, number: -1, abs: true))
      line(b, c, stroke: red)
    
      cetz.angle.angle(b, a, c, label: $ dot $)
      
      fill(black)
      circle(b)
      circle(c)
    
      content(a, $ g $, anchor: "bottom")
      content(b, $ F $, anchor: "bottom")
      content(c, $ P $, anchor: "left")
    }), [
      Gesucht ist der Abstand von Punkt $P$ zur Geraden $g$.
      *Vorgehen:* \
      + Lot von $P$ auf $g$ mit $(arrow(p) - arrow(f)) dot arrow(u) = 0$ mit 
        unbekanntem Punkt $F$ auf der Geraden aufstellen
      + ...
      + ...
    ])})
```

![grafik](https://github.com/johannes-wolf/cetz/assets/519002/00ba444b-7a47-4650-95e5-5207c20c4b99)


</details>

**Reply by Loewe1000** [reply]
Wow, thank you very much for the quick and detailed answer.

That helps me a lot!

Thank you also for CeTZ in general. I love it.
